### PR TITLE
Netcdf append

### PR DESCRIPTION
**Pull request by radosuav**
Follow on from https://github.com/scipy/scipy/pull/4960

Fixes two issues which were present when appending to an existing NetCDF file.

First commit fixes the ValueError which appears when in-place resize is used on a numpy array storing a NetCDF record variable. The exception is raised since the array might not be single-segment. The in-place resize is still tried first since it is probably faster than copy resize, whenever it is possible.

Second commit ensures that "data" field is not saved as a user defined attribute, since it is used to access the variable data.

In reply to pv comments in https://github.com/scipy/scipy/pull/4960, .astype is needed at least on line 994 but I would prefer to as well keep it on line 517 just in case.

### CODE REVIEW

File: scipy/io/tests/test_netcdf.py [hunk 1]
```diff
@@ -317,6 +317,57 @@ def test_open_append():
         f.close()
 
 
+def test_append_recordDimension(): 
+    dataSize = 100   
+    cwd = os.getcwd()
```
Comment (pv) [inline scipy/io/tests/test_netcdf.py]: `with in_tempdir():`

File: scipy/io/netcdf.py [hunk 2]
```diff
@@ -509,7 +509,12 @@ def _write_var_data(self, name):
             # Handle rec vars with shape[0] < nrecs.
             if self._recs > len(var.data):
                 shape = (self._recs,) + var.data.shape[1:]
-                var.data.resize(shape)
+                # Resize in-place does not always work since 
+                # the array might not be single-segment                              
+                try:
+                    var.data.resize(shape)
+                except ValueError:
+                    var.data = np.resize(var.data, shape).astype(var.data.dtype)
```
Comment (pv) [inline scipy/io/netcdf.py]: Suggest writing it as `var.__dict__['data'] = ...` so that you don't need to add a special case in `__setattr__`.

**Comment by radosuav** [comment]
Thanks for the hints! I updated as suggested.

**Comment by rgommers** [comment]
LGTM, merged. Thanks @radosuav

**Pull request merged**